src/x86/runtime.rs

```diff
@@ -0,0 +1,120 @@
+#[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+#[macro_export]
+macro_rules! __unstable_detect_feature_x86 {
```

Comment:
It may actually be simpler if this is defined on all platforms and just returns `false` for all features?

Reply:
What do you mean? On ARM we won't probably need such a macro because all the feature detection will happen on the parent macro (ARM features can only be portably detected at compile-time).

Reply:
I can refactor this somewhere else, but I think it would be better to make compilation fail on ARM if one tries to do run-time feature detection for, e.g., SSE4.

Reply:
I'd just imagine something like:

```rust

#[cfg(x86)]
macro_rules! __unstable_detect_feature { ... }
#[cfg(arm)]
macro_rules! __unstable_detect_feature { ... }
#[cfg(everything_else)]
macro_rules! __unstable_detect_feature { ... }
```

Reply:
Ah! I see, I can do that!

Reply:
This will look like this though: 

```rust
#[cfg(all(target_arch = "x86", target_arch = "x86_64"))]
macro_rules! __unstable_detect_feature { ... }
#[cfg(all(target_arch = "arm", target_arch = "aarch64"))]
macro_rules! __unstable_detect_feature { ... }
// ...
#[cfg(not(any(target_arch = "x86", target_arch = "x86_64", target_arch = "arm", target_arch = "aarch64", ...)))]
macro_rules! __unstable_detect_feature { ... }
```

Reply:
Fixed.